src/Nox.Generator/Common/NoxSolutionCodeGeneratorState.cs

```diff
@@ -19,6 +21,21 @@ public NoxSolutionCodeGeneratorState(NoxSolution noxSolution)
         public string PersistenceNameSpace => $"{RootNameSpace}.Infrastructure.Persistence";
         public string ODataNameSpace => $"{RootNameSpace}.Presentation.Api.OData";
         public string Events => $"{RootNameSpace}.Application.Events";
+        public string GetEntityTypeFullName(string entityName) => $"{DomainNameSpace}.{entityName}";
+        // TODO: could be optimized
+        public Type? GetEntityType(string entityName)
```

Comment:
This will be always in the EntryAssembly, so we can receive on the constructor of this class the EntryAssembly, maybe call it ClientAssembly?, and search will be much more fast. In the constructor because tests will have a different assembly